Source/pfile.cpp

```diff
@@ -243,6 +247,42 @@ bool ArchiveContainsGame(HANDLE hsArchive)
 	return IsHeaderValid(hdr);
 }
 
+bool CompareSaves(std::string actualSavePath, std::string referenceSavePath)

```

Comment:
I'm having a hard time seeing why this should live in pfile. Also instead of using a full blown save for comparison I think it would be better to just have a hash generated by `picosha2`

Reply:
This are two topics.

### Binary format on disk:
In my first version I simply tried to write the save game and compare the written file with the original. Sadly this doesn't worked. So I thought someone adds data was always changes (random data or a timestamp). Then I changed save game creation that less and less data was written. But at the end it seams that stormlib or the codec functions add random bytes.
That's why I ended up with comparing the uncompressed content of the mpq. And for that I need functions like `ReadArchive` or `GetPermSaveNames` that all live in `pfile.cpp`. 🙂 
I think the same problem also applies for hash generation.

### Hash comparsion
In the future I think this is a good thing. Cause it's decreases the needed size massively. Currently I would stick with the save game, cause it allows better diagnostic what changed (cause you can compare content) and I'm currently not sure if I compare enough data (are some files in the mpq missing?).